src/test/java/org/opensearch/security/auditlog/AbstractAuditlogiUnitTest.java

```diff
@@ -83,34 +84,46 @@ protected void setupStarfleetIndex() {
         rh.keystore = keystore;
     }
 
-    protected boolean validateMsgs(final Collection<AuditMessage> msgs) {
-        boolean valid = true;
+    protected void validateMsgs(final Collection<AuditMessage> msgs) throws Exception {
         for (AuditMessage msg : msgs) {
-            valid = validateMsg(msg) && valid;
+            try {
+                validateMsg(msg);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(msg + " errored because of following validateMsg error: " + e.getMessage(), e);
+            } catch (JsonMappingException e) {
+                throw e;
+            }
         }
-        return valid;
-    }
 
-    protected boolean validateMsg(final AuditMessage msg) {
-        return validateJson(msg.toJson()) && validateJson(msg.toPrettyString());
     }
 
-    protected boolean validateJson(final String json) {
-
-        if (json == null || json.isEmpty()) {
-            return false;
+    protected void validateMsg(final AuditMessage msg) throws Exception {
+        try {
+            validateJson(msg.toJson());
+        } catch (IllegalArgumentException e) {
```

Comment:
You can remove these catch blocks since the method throws `Exception` and you end up rethrowing the exception anyways

Reply:
I wanted to build on to the exception message as it travels up the method stack trace, do you think this is required?

Reply:
Or if this is good, how can I do that without a catch block?